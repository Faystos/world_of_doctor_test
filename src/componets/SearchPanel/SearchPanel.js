import React, { Component } from 'react';
import ApiService from '../../services/services';
import Loader from '../Loader';

export default class SearchPanel extends Component {
  apiService = new ApiService();

  state = {
    data: null,    
    load: true,
    error: false,
    searchValue: ''
  } 

  onLoadData = (value) => {
    this.apiService.sortCountries(value)
      .then(data => this.setState({ data, load: false}))
      .catch(err => console.log(err));
  }

  onValueSerh = value => {
    this.setState(({searchValue}) => {
      return {searchValue: value};
    }); 
    this.onLoadData(value);       
  }  

  renderItem(arr, index, searchValue) {
    const reg = new RegExp("^[a-zA-Z]+$");    
    let items = [];
    let quantity = index > arr.length ? arr.length : index;    
    for (let i = 0; i < quantity; i++) {      
      items.push(<span className="matchItem" key={arr[i].countryID}>{`${i + 1}. ${arr[i].countryName}`}</span>);
    }
    if (!items.length) items = 'Такого значения не существует';
    if (!reg.test(searchValue)) items = 'Только латинские символы';
    return items;      
  }

  render () {
    const { data, load, searchValue } = this.state;
    const viewDropPownMatch = searchValue ? <DropPownMatch data = { data } load = { load } renderItem = { this.renderItem } searchValue= { searchValue }/> : null;
    
     
    
    return(
      <section className="searchPanel">
        <div className = "blockInp">
          <SearchInput onValueSerh = { this.onValueSerh }/>
          <InputSelect />
          
        </div>
        
        { viewDropPownMatch }
        
        
        
      </section>      
    );
  }
}

// ********************************************************************

class SearchInput extends Component {
  state = {
    value: '',
  }

  changeSearchInt = ({target}) => {
    let value = target.value;  
    this.setState({ value });
    this.props.onValueSerh(value);
  }

  render() {
    const { value } = this.state;
    
    return(
      <div className="searchInput">
        <input type= "text" placeholder="input country" value = {value} onChange = {this.changeSearchInt}/>
      </div>
    );
  }  
}

// ************************************************************************

const DropPownMatch = ({ data, load, renderItem, searchValue }) => {    
  const hasData = (data && !load);
  let viewLoader = load ? < Loader /> : null;
  let viewItem = hasData ? renderItem(data, 10, searchValue) : null;      
  return(
    <div className="blockMatch">
      { viewLoader }
      { viewItem }
    </div>
  );  
}

const InputSelect = () => {

  return(
    <React.Fragment>
      <select>
        <option>10</option>
        <option>50</option>
        <option>100</option>
      </select>
    </React.Fragment>
  );
}