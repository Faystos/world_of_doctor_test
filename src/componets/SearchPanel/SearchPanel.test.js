import React from 'react';
import renderer from 'react-test-renderer';
import SearchPanel from './SearchPanel';
import ReactTestUtils from 'react-dom/test-utils';

test('test SearchPanel', () => {
  const component = renderer.create(
    <SearchPanel />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();  
});

describe('SearchPanel Component', () => {
  it('has an section tag', () => {
    const component = ReactTestUtils.renderIntoDocument(<SearchPanel/>);    
    let section = ReactTestUtils.findRenderedDOMComponentWithTag(
     component, 'section');
   });   
  
   it('is wrapped inside a searchPanel class', () => {
    const component = ReactTestUtils.renderIntoDocument(<SearchPanel/>);    
    let classN = ReactTestUtils.findRenderedDOMComponentWithClass(
    component, 'searchPanel');
   });

   it('is wrapped inside a blockInp class', () => {
    const component = ReactTestUtils.renderIntoDocument(<SearchPanel/>);    
    let classN = ReactTestUtils.findRenderedDOMComponentWithClass(
    component, 'blockInp');
   });
});

