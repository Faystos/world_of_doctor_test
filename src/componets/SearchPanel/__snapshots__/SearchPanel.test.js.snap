// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SearchPanel 1`] = `
<section
  className="searchPanel"
>
  <div
    className="blockInp"
  >
    <div
      className="blockSearchInput"
    >
      <input
        className="searchInput"
        id="search"
        onChange={[Function]}
        placeholder="input country"
        type="text"
        value=""
      />
      <label
        className="searchLabel"
        htmlFor="name"
      >
        Country
      </label>
    </div>
    <select
      className="inpSelect"
      onChange={[Function]}
    >
      <option
        value="10"
      >
        10
      </option>
      <option
        value="20"
      >
        20
      </option>
      <option
        value="50"
      >
        50
      </option>
    </select>
  </div>
</section>
`;
